### PR TITLE
DeriveKey Scrypt unit test

### PR DESCRIPTION
**Pull request by shargon**
The number of iterations is set to only to 1, last NIST recommendation (from 2016) is 10.000 (https://pages.nist.gov/800-63-3/sp800-63b.html#sec5).

### CODE REVIEW

File: neo/Cryptography/SCrypt.cs [hunk 1]
```diff
@@ -261,7 +266,7 @@ public unsafe static byte[] DeriveKey(byte[] password, byte[] salt, int N, int r
             var mac = new HMACSHA256(password);
 
             /* 1: (B_0 ... B_{p-1}) <-- PBKDF2(P, S, 1, p * MFLen) */
-            PBKDF2_SHA256(mac, password, salt, salt.Length, 1, Ba, p * 128 * r);
+            PBKDF2_SHA256(mac, password, salt, salt.Length, PBKDF2_Iterations, Ba, p * 128 * r);
```
Comment (erikzhang) [inline neo/Cryptography/SCrypt.cs]: The comments should be changed too.

**Comment by erikzhang** [comment]
This does not change the results, right?

**Comment by shargon** [comment]
Yes, this change the results of the output hash

**Comment by erikzhang** [comment]
I think it is not compatible with scrypt. In [RFC 7914](https://tools.ietf.org/html/rfc7914), it is fixed at `1`.

**Comment by shargon** [comment]
Because scrypt add extra security layer (and this method is used only with scrypt), maybe is not needed. If you want i can remove all changes except the unit test

**Comment by erikzhang** [comment]
Yes, please.

File: neo.UnitTests/UT_Scrypt.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        public void DeriveKeyTest()
+        {
+            int N = 16384, r = 8, p = 8;
+
+            var date = DateTime.UtcNow;
+            var derivedkey = SCrypt.DeriveKey(new byte[] { 0x01, 0x02, 0x03 }, new byte[] { 0x04, 0x05, 0x06 }, N, r, p, 64).ToHexString();
+            var elapsed = (DateTime.UtcNow - date);
+
+            Assert.AreEqual("2bb9c7bb9c392f0dd37821b76e42b01944902520f48d00946a51e72c960fba0a3c62a87d835c9df10a8ad66a04cdf02fbb10b9d7396c20959f28d6cb3ddfdffb", derivedkey);
+            Assert.IsTrue(elapsed.TotalSeconds > 1);
```
Comment (erikzhang) [inline neo.UnitTests/UT_Scrypt.cs]: This will fail when we use a super fast CPU.

Comment (shargon) [reply neo.UnitTests/UT_Scrypt.cs]: Then we should increase the scrypt values

Comment (vncoelho) [reply neo.UnitTests/UT_Scrypt.cs]: EHuaheauea, maybe just report the value is enough, Shargon, no?

Comment (igormcoelho) [reply neo.UnitTests/UT_Scrypt.cs]: Is it intended to spend at least one second?

Comment (shargon) [reply neo.UnitTests/UT_Scrypt.cs]: Should be hard to bruteforce it

Comment (vncoelho) [reply neo.UnitTests/UT_Scrypt.cs]: And it also means 1more sec of UT...aejuaheuauhea

Comment (erikzhang) [reply neo.UnitTests/UT_Scrypt.cs]: I think tests should be deterministic.

**Pull request merged**